appcues/src/main/java/com/appcues/data/mapper/MapperExt.kt

```diff
@@ -1,18 +1,16 @@
 package com.appcues.data.mapper
 
 import com.appcues.data.remote.response.action.ActionResponse
-import com.appcues.data.remote.response.trait.TraitResponse
+import com.appcues.trait.ExperienceTrait
+import com.appcues.trait.ExperienceTrait.ExperienceTraitLevel
 import java.util.UUID
 
-internal fun List<TraitResponse>.mergeTraits(other: List<TraitResponse>): List<TraitResponse> {
-    return mutableListOf<TraitResponse>().apply {
+internal fun List<ExperienceTrait>.mergeTraits(others: List<ExperienceTrait>, level: ExperienceTraitLevel): List<ExperienceTrait> {
```

Comment:
instead of dealing with TraitResponse, we are now using ExperienceTrait, so we can check its "level" to know if we should propagate that trait to groups, steps